app/code/Magento/CatalogWidget/Block/Product/ProductsList.php

```diff
@@ -162,6 +170,7 @@ public function __construct(
     ) {
         $this->productCollectionFactory = $productCollectionFactory;
         $this->catalogProductVisibility = $catalogProductVisibility;
+        $this->optionsDataViewModel = $optionsDataViewModel;
```

Comment:
In order to make this Backward compatible new params in Constructor should be null and handle with ObjectManager:

```suggestion
        $this->optionsDataViewModel = $optionsDataViewModel ?: ObjectManager::getInstance()->get(OptionsData::class);
```